static/src/javascripts/projects/common/modules/loyalty/save-for-later.js

```diff
@@ -0,0 +1,90 @@
+define([
+    'qwery',
+    'bonzo',
+    'bean',
+    'common/utils/_',
+    'common/utils/config',
+    'common/utils/mediator',
+    'common/modules/identity/api'
+], function (
+    qwery,
+    bonzo,
+    bean,
+    _,
+    config,
+    mediator,
+    identity
+) {
+    function SaveForLater() {
+        this.saveLinkHolder = qwery('.meta__save-for-later')[0];
+        this.userData = null;
+        this.encodedPageUrl = encodeURIComponent(document.location.href);
+        this.$saver = bonzo(this.saveLinkHolder);
+        this.savedArticlesUrl = config.page.idUrl + '/saved-articles';
+    }
+
+    SaveForLater.prototype.init = function () {
+        if (identity.isUserLoggedIn()) {
+            this.getSavedArticles();
+        } else {
+            var url = config.page.idUrl + '/prefs/save-content?returnUrl=' + this.encodedPageUrl + '&shortUrl=' + config.page.shortUrl;
+            this.$saver.html(
+                '<a href="' + url + ' "data-link-name="meta-save-for-later" data-component=meta-save-for-later">Save for later</a>'
+            );
+        }
+    };
+
+    SaveForLater.prototype.getSavedArticles = function () {
+        var self = this,
+            notFound  = [{message:'Not found', description:'Resource not found'}];
+
+        identity.getUsersSavedDocuments().then(
+            function success(resp) {
+                if (resp.status === 'error') {
+                    if (JSON.stringify(notFound) ===  JSON.stringify(resp.errors)) {
```

Comment:
I think it's better to compare `resp.errors[0].message === notFound.message && resp.errors[0].description === notFound.description`.
You can't guarantee that JSON.stringify maintains the same order of properties for two different objects.